File: code.py
import gspread
from oauth2client.service_account import ServiceAccountCredentials
from datetime import datetime

def submit_form_data(data):
    try:
        # กำหนด scope สำหรับ Google Sheets API
        scope = ['https://spreadsheets.google.com/feeds',
                 'https://www.googleapis.com/auth/drive']

        # โหลด credentials จากไฟล์ JSON
        creds = ServiceAccountCredentials.from_json_keyfile_name('credentials.json', scope)

        # เชื่อมต่อกับ Google Sheets API
        client = gspread.authorize(creds)

        # เปิดสเปรดชีตและชีต
        sheet = client.open_by_key('1V5W5JJMuPQyy6pqerd7saHh89HAeXUH6Tpfh52dpsQo').sheet1  # หรือใช้ sheet.worksheet('Sheet1')

        # เพิ่มข้อมูลเป็นแถวใหม่
        row = [
            datetime.now().strftime("%Y-%m-%d %H:%M:%S"),  # Timestamp
            data['name'],
            data['department'],
            data['phone'],
            data['date1'],
            data['date2'],
            data['count'],
            data['room']
        ]
        sheet.append_row(row)

        return {'status': 'success', 'message': 'บันทึกข้อมูลสำเร็จ!'}

    except Exception as e:
        return {'status': 'error', 'message': str(e)}

# ตัวอย่างการใช้งาน
data = {
    'name': 'John Doe',
    'department': 'IT',
    'phone': '0812345678',
    'date1': '2023-10-26',
    'date2': '2023-10-27',
    'count': '10',
    'room': 'พัก'
}

result = submit_form_data(data)
print(result)

File: code.js
function doGet(e) {
  // เมื่อมีการเปิด Web App (URL ของ Apps Script)
  // จะแสดงหน้า index.html ผ่าน HTML Service
  return HtmlService.createTemplateFromFile('index')
    .evaluate()
    .setTitle('แบบฟอร์มจองห้องประชุม');
    // หากต้องการฝังใน iFrame ที่อื่น อาจต้องใช้ .setXFrameOptionsMode(HtmlService.XFrameOptionsMode.ALLOWALL)
}

/**
 * ฟังก์ชันที่หน้า HTML เรียกผ่าน google.script.run
 * เพื่อบันทึกข้อมูลลง Google Sheets
 */
function submitFormData(data) {
  try {
    // 1) เปิดสเปรดชีตตาม Spreadsheet ID
    //    (ดูจาก URL ของ Google Sheets)
    var sheet = SpreadsheetApp.openById("1QRwybvcC_98EL9si__zxfyeo_X7fZBV7tknrHoCVhqw") 
                              .getSheetByName("Sheet1"); // ปรับชื่อชีตตามต้องการ
    
    // 2) เพิ่มข้อมูลเป็นแถวใหม่
    //    เก็บ Timestamp (new Date), date, location, details
    sheet.appendRow([
      new Date(),        // Timestamp
      data.name,
      data.department,
      data.phone,
      data.date1,         // วันที่
      data.date2,         // วันที่
      data.count,     // จำนวน
      data.room       // ห้องพัก
    ]);

    // 3) ส่งสถานะ "success" กลับไป
    return {
      status: 'success',
      message: 'บันทึกข้อมูลสำเร็จ!'
    };

  } catch (error) {
    // หากมีปัญหา ส่งสถานะ "error" กลับ
    return {
      status: 'error',
      message: error.message
    };
  }
}

/**
 * ฟังก์ชันช่วย (Include) สำหรับกรณีมีหลายไฟล์ HTML แยก
 * กรณีนี้ถ้าไฟล์ index.html ไฟล์เดียว อาจไม่จำเป็นใช้ก็ได้
 */
function include(filename) {
  return HtmlService.createTemplateFromFile(filename).getRawContent();
}
